code/game/machinery/computer/crew.dm

```diff
@@ -0,0 +1,180 @@
+/obj/machinery/computer/crew
+	name = "Crew monitoring computer"
+	desc = "Used to monitor active health sensors built into most of the crew's uniforms."
+	icon_state = "crew"
+	use_power = 1
+	idle_power_usage = 250
+	active_power_usage = 500
+	circuit = "/obj/item/weapon/circuitboard/crew"
+	var/list/tracked = list(  )
+	var/track_special_role
+
+	light_color = LIGHT_COLOR_BLUE
+	light_range_on = 2
+
+
+/obj/machinery/computer/crew/New()
+	tracked = list()
+	..()
+
+
+/obj/machinery/computer/crew/attack_ai(mob/user)
+	attack_hand(user)
+	ui_interact(user)
+
+
+/obj/machinery/computer/crew/attack_hand(mob/user)
+	add_fingerprint(user)
+	if(stat & (BROKEN|NOPOWER))
+		return
+	ui_interact(user)
+
+
+/obj/machinery/computer/crew/update_icon()
+
+	if(stat & BROKEN)
+		icon_state = "crewb"
+	else
+		if(stat & NOPOWER)
+			src.icon_state = "c_unpowered"
+			stat |= NOPOWER
+		else
+			icon_state = initial(icon_state)
+			stat &= ~NOPOWER
+
+
+/obj/machinery/computer/crew/Topic(href, href_list)
+	if(..()) return
+	if (src.z > 6)
+		usr << "<span class='danger'>Unable to establish a connection: </span>You're too far away from the station!"
```

Comment:
I disagree.

Silently eating blatant errors is always worse than a 600 MiB runtime log.